SingularityUI/app/selectors/requests.es6

```diff
@@ -0,0 +1,101 @@
+import { createSelector } from 'reselect';
+
+import Utils from '../utils';
+
+const getStarred = (state) => state.ui.starred;
+const getRequests = (state) => state.api.requests;
+const getUser = (state) => state.api.user;
+
+export const combineStarredWithRequests = createSelector(
+  [getStarred, getRequests],
+  (starredData, requestsAPI) => {
+    // TODO: make this use two sorted lists
+    // to combine the data
+    // because this is O(awful)
+    return requestsAPI.data.map((r) => {
+      if (starredData.indexOf(r.request.id) > -1) {
+        return {
+          ...r,
+          starred: true
+        };
+      }
+      return r;
```

Comment:
- consider simplifying to `requestsAPI.data.map((r) => ({...r, starred: (starredData.indexOf(r.request.id) > -1)}))`
- you can store starredData as a Set to speed up lookups